### PR TITLE
Add examples for options with descriptions for combobox and dropdown

### PR DESCRIPTION
**Pull request by ThusharaJ07**

### CODE REVIEW

File: packages/core/stories/combo-box/combo-box.stories.tsx [hunk 1]
```diff
@@ -653,6 +655,54 @@ MultiplePillsTruncated.args = {
...
+      onChange={handleChange}
+      onSelectionChange={handleSelectionChange}
+      value={value}
+      valueToString={(person) => person.displayName}
+    >
+      {options.map((person) => (
+        <Option key={person.id} value={person}>
+          <StackLayout gap={0.25} align="start">
+            <Text variant="primary">{person.firstName}</Text>
+            <Label variant="secondary">{person.lastName}</Label>
```
Comment (joshwooding) [inline packages/core/stories/combo-box/combo-box.stories.tsx:698]: Label isn't semantically correct here we could probably just use:

```suggestion
            <Text variant="secondary">{person.lastName}</Text>
```

But we should also use the color prop

File: packages/core/stories/combo-box/combo-box.stories.tsx [hunk 2]
```diff
@@ -653,6 +655,54 @@ MultiplePillsTruncated.args = {
   truncate: true,
 };
 
+export const SecondaryLabel: StoryFn<ComboBoxProps<Person>> = (args) => {
```
Comment (joshwooding) [inline packages/core/stories/combo-box/combo-box.stories.tsx:658]: The Storybook and site example should be the same

**Comment by origami-z** [comment]
In #3431, gap between labels are `0.5`, maybe this should be updated as well?

**Comment by ThusharaJ07** [comment]
> In #3431, gap between labels are `0.5`, maybe this should be updated as well?

updated the gap to 0.5

**Review by origami-z — DISMISSED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by shey-v — APPROVED** [review]

**Pull request merged**